### DISCUSSION TITLE
How to return 204 status code in route.js?

### DISCUSSION THREAD

**Discussion by kmvan** [vercel/next.js#51475] [App Router] [answered]
### Summary

```js
// app/test/route.js
export async function POST() {
  return NextResponse.json(null,{
    status: 204
  })
}
```
I got an error: `TypeError: Response constructor: Invalid response status code 204`.
Any idea?

**Comment by devjiwonchoi** [comment]
`NextResponse` does not have that option at the moment, and is going to take some time for that to be added. I'm going to open a PR for that feature and will let you know if that happens.

**Reply by kmvan** [reply]
Thank you. I will follow closely.

**Comment by icyJoseph** [comment, marked as answer]
Just use the `new Response` form:

For example for a 200,

```js
export async function GET(request: Request) {
  return new Response('Hello, Next.js!', {
    status: 200,
  })
}
```

Or a 204:

```ts
export async function GET(request: Request) {
  return new Response(null, {
    status: 204,
  })
}
```

**Reply by kmvan** [reply]
Using Response is also a very good idea.

**Reply by donprasetiyo** [reply]
This doesn't work for me. I'm using Axios.

**Reply by kmvan** [reply]
What is your status code?

**Reply by bonface221** [reply]
@kmvan  it is working on my end

**Reply by homocodian** [reply]
your first parameter should be null while using 204 status code

```js
export async function GET(request: Request) {
  return new Response(null, {
    status: 204,
  })
}
```

**Not this**

```js
export async function GET(request: Request) {
  return new Response('Hello, Next.js!', {
    status: 204,
  })
}
```

**Reply by hakan-berat** [reply]
@homocodian You saved my life, thanks!!!

**Reply by homocodian** [reply]
> @homocodian You saved my life, thanks!!!

Glad I could help!

**Reply by zecka** [reply]
This seems fixes issue on dev server but after build i still get a 500 error with following error

⨯ TypeError: Response constructor: Invalid response status code 204

Other people seems to have same issue here: https://github.com/vercel/next.js/issues/49005

**Reply by icyJoseph** [reply]
> This seems fixes issue on dev server but after build i still get a 500 error with following error
> 
> ⨯ TypeError: Response constructor: Invalid response status code 204
> 
> Other people seems to have same issue here: https://github.com/vercel/next.js/issues/49005

Could it be that the route is statically optimized, though you return a 204? Try adding `export const dynamic = "force-dynamic"` to the route handler.
